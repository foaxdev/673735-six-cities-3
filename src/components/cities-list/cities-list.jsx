import React from "react";
import PropTypes from "prop-types";

export const CitiesList = ({cities, currentCity, onCityClick}) => {

  const listOfCities = cities.map((city) =>
    <li className="locations__item" key={city}>
      <a className={city === currentCity ? `locations__item-link tabs__item tabs__item--active` : `locations__item-link tabs__item`} href="#" onClick={() => onCityClick(city)}>
        <span>{city}</span>
      </a>
    </li>
  );

  return (
    <ul className="locations__list tabs__list">
      {listOfCities}
    </ul>
  );
};

CitiesList.propTypes = {
  cities: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  currentCity: PropTypes.string.isRequired,
  onCityClick: PropTypes.func.isRequired
};
