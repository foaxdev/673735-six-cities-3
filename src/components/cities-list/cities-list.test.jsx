import React from 'react';
import renderer from 'react-test-renderer';
import {CitiesList} from "./cities-list";

const TEST_HANDLER = () => {};
const TEST_CITIES = [`Japan`, `Korea`, `China`, `Singapore`];
const TEST_CURRENT_CITY = `Japan`;

it(`Should render Cities List correctly`, () => {
  const tree = renderer
    .create(
        <CitiesList
          onCityClick={TEST_HANDLER}
          cities={TEST_CITIES}
          currentCity={TEST_CURRENT_CITY}
        />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
