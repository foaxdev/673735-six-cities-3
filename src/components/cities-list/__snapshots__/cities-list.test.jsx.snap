// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Cities List correctly 1`] = `
<ul
  className="locations__list tabs__list"
>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item tabs__item--active"
      href="#"
      onClick={[Function]}
    >
      <span>
        Japan
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        Korea
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        China
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        Singapore
      </span>
    </a>
  </li>
</ul>
`;
